ursula_cli/shell.py

```diff
@@ -195,7 +195,8 @@ def _run_heat(args, hot):
     CREDS = {
         'username': os.environ['OS_USERNAME'],
         'password': os.environ['OS_PASSWORD'],
-        'tenant_name': os.environ['OS_TENANT_NAME'],
+        'tenant_name': (os.environ.get('OS_TENANT_NAME') or
+                        os.environ['OS_PROJECT_NAME']),
```

Comment:
This is where I'm getting confused: this is a hacky way of getting out of doing the explicit check for either `OS_TENANT_NAME` or `OS_PROJECT_NAME`.  Though I think that bubbling up errors through this dict literal construction via `KeyError`s isn't ideal either.

So my recommendation is that we should do explicit checks.  I'll offer a patch.